src/main/java/ua/com/juja/microservices/keepers/slackbot/controller/KeepersSlackCommandController.java

```diff
@@ -18,77 +26,136 @@
-                                 @RequestParam("text") String text) {
-
-        logger.debug("Received slash command KeeperAdd: from user: [{}] command: [{}] token: [{}]",
-                fromUser, text, token);
-
-        if (!token.equals(slackToken)) {
-            logger.warn("Received invalid slack token: [{}] in command KeeperAdd for user: [{}]", token, fromUser);
-            return new RichMessage("Sorry! You're not lucky enough to use our slack command.");
+    public void addKeeper(@RequestParam("token") String token,
+                         @RequestParam("user_name") String fromUser,
+                         @RequestParam("text") String text,
+                         @RequestParam("response_url") String responseUrl,
+                         HttpServletResponse httpServletResponse) throws IOException {
+        try {
+            LOGGER.debug("Received slash command KeeperAdd: from user: [{}] command: [{}] token: [{}] responseUrl: [{}]",
+                    fromUser, text, token, responseUrl);
+
+            if (!token.equals(slackToken)) {
+                LOGGER.warn("Received invalid slack token: [{}] in command KeeperAdd for user: [{}]", token, fromUser);
+                sendQuickResponse(httpServletResponse, SORRY_MESSAGE);
+            } else {
+                sendQuickResponse(httpServletResponse, IN_PROGRESS);
+                String response = keeperService.sendKeeperAddRequest(fromUser, text);
+                LOGGER.info("KeeperAdd command processed : user: [{}] text: [{}] and sent response into slack: [{}]",
+                        fromUser, text, response);
+                sendDelayedResponse(responseUrl, response);
+            }
+        } catch (BaseBotException bex){
+            sendBBEMessage(responseUrl, bex);
+        } catch (Exception ex){
+            sendEMessage(responseUrl, ex);
         }
-
-        String response = keeperService.sendKeeperAddRequest(fromUser, text);
-
-        logger.info("Keeper command processed : user: [{}] text: [{}] and sent response into slack: [{}]",
-                fromUser, text, response);
-
-        return new RichMessage(response);
     }
 
     @PostMapping(value = "/dismiss", consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
-    public RichMessage dismissKeeper(@RequestParam("token") String token,
-                                     @RequestParam("user_name") String fromUser,
-                                     @RequestParam("text") String text) {
-
-        logger.debug("Received slash command KeeperDismiss: from user: [{}] command: [{}] token: [{}]",
-                fromUser, text, token);
-
-        if (!token.equals(slackToken)) {
-            logger.warn("Received invalid slack token: [{}] in command KeeperDismiss for user: [{}]", token, fromUser);
-            return new RichMessage("Sorry! You're not lucky enough to use our slack command.");
+    public void dismissKeeper(@RequestParam("token") String token,
```

Comment:
please rename to 'deactivateKeeper' - method, url, api

Reply:
ok

Reply:
fixed